packages/compose/src/hooks/use-copy-on-click/index.js

```diff
@@ -9,32 +9,40 @@ import Clipboard from 'clipboard';
 import { useRef, useEffect, useState } from '@wordpress/element';
 import deprecated from '@wordpress/deprecated';
 
+/* eslint-disable jsdoc/no-undefined-types */
 /**
  * Copies the text to the clipboard when the element is clicked.
  *
  * @deprecated
  *
- * @param {Object}          ref     Reference with the element.
- * @param {string|Function} text    The text to copy.
- * @param {number}          timeout Optional timeout to reset the returned
+ * @param {import('react').RefObject<string | Element | NodeListOf<Element>>} ref Reference with the element.
```

Comment:
Shouldn't we import from `@wordpress/element`?

Reply:
`@wordpress/element` doesn't export all the types, just the functions. We import the types directly from `react` everywhere else as well so it's just following the already established pattern.

Reply:
I also prefer importing the types in a comment above to leave more room for parameter descriptions

Reply:
Ah, I think there's lots of places where we import from `@wordpress/element`. Why don't we export the patterns there?

Reply:
Frankly there are a lot of types to export if we were to export them from `@wordpress/element`. This would increase the maintenance cost of that already quite non-standard package. We've also already accepted importing them directly from `react` in this case, so it would be a departure from what's already been going on.

I think @sirreal had some opinions about this that he might want to chime in with.

Reply:
Re-exporting React types from `@wordpress/element` was discussed here: https://github.com/WordPress/gutenberg/issues/21767

From memory:
- Re-exporting the types in JSDoc is extremely costly. Remember that we'd need to completely reproduce in JSDoc anything that accepts a type argument, e.g.
- We could re-evaluate that now that some TypeScript language is allowed. However, I don't believe we have a "full" re-export of React, so we'd still need to re-export every single type we want.
- Using types from React is "free". I don't see downsides to using it for the types especially considering the costs of the alternative.